#!/usr/bin/env python
PACKAGE = "franka_ros_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

"""
    Joint Position Controller (position_joint_position_controller) smoothing parameter. Increasing the value gives higher smoothing (smaller change in target position per controller update step).
"""
gen.add("position_joint_delta_filter", double_t, 0, "Filter parameter that dictates how smoothly subsequent joint position commands are smoothed. Also responsible for speed of joint position execution.", 0.3, 0.1, 1.0) # (default, min, max)
gen.add("velocity_joint_delta_filter", double_t, 0, "Filter parameter that dictates how smoothly subsequent joint velocity commands are smoothed. Also responsible for acceleration of joints.", 0.3, 0.1, 1.0) # (default, min, max)


"""
    Joint positions controller gains
"""
controller_gains = gen.add_group("Controller_Gains")

controller_gains.add("j1_k", double_t, 0, "Stiffness parameter of joint 1", 1200, 200, 1500) # (default, min, max)
controller_gains.add("j2_k", double_t, 0, "Stiffness parameter of joint 2", 1000, 200, 1500)
controller_gains.add("j3_k", double_t, 0, "Stiffness parameter of joint 3", 1000, 200, 1500)
controller_gains.add("j4_k", double_t, 0, "Stiffness parameter of joint 4", 800, 200, 1500)
controller_gains.add("j5_k", double_t, 0, "Stiffness parameter of joint 5", 300, 100, 1000)
controller_gains.add("j6_k", double_t, 0, "Stiffness parameter of joint 6", 200, 75, 500)
controller_gains.add("j7_k", double_t, 0, "Stiffness parameter of joint 7", 50, 20, 200)

controller_gains.add("j1_d", double_t, 0, "Damping parameter of joint 1", 50, 0, 200) # (default, min, max)
controller_gains.add("j2_d", double_t, 0, "Damping parameter of joint 2", 50, 0, 200)
controller_gains.add("j3_d", double_t, 0, "Damping parameter of joint 3", 50, 0, 200)
controller_gains.add("j4_d", double_t, 0, "Damping parameter of joint 4", 20, 0, 200)
controller_gains.add("j5_d", double_t, 0, "Damping parameter of joint 5", 20, 0, 200)
controller_gains.add("j6_d", double_t, 0, "Damping parameter of joint 6", 20, 0, 200)
controller_gains.add("j7_d", double_t, 0, "Damping parameter of joint 7", 10, 0, 200)

exit(gen.generate(PACKAGE, "controller_configurations", "joint_controller_params"))
